cmd/duffle/push.go

```diff
@@ -0,0 +1,112 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/deislabs/duffle/pkg/duffle/home"
+	"github.com/deislabs/duffle/pkg/reference"
+
+	"github.com/docker/cli/cli/config"
+	"github.com/docker/cnab-to-oci/remotes"
+	"github.com/spf13/cobra"
+)
+
+type pushCmd struct {
+	inputBundle        string
+	home               home.Home
+	bundleIsFile       bool
+	targetRef          string
+	insecureRegistries []string
+	allowFallbacks     bool
+}
+
+func newPushCmd(out io.Writer) *cobra.Command {
+	const usage = `Pushes a CNAB bundle to an OCI repository.`
+	const pushDesc = `
+Pushes a CNAB bundle to an OCI registry by pushing all container
+images referenced in the bundle to the target repository (all images are
+pushed to the same repository, and are referenceable through their digest).
```

Comment:
Where are images "pushed" _from_? Are they copied direct from their remote repositories or do they have to be present in a docker daemon?

Reply:
Yes, they are directly copied from the remote (and actually, as #734 tracks, if the images are only present in the local daemon, this will fail)..